src/main/java/com/speaker/config/kafka/CustomSerializer.java

```diff
@@ -0,0 +1,47 @@
+package com.speaker.config.kafka;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.speaker.dto.MessageDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Serializer;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+@Slf4j
+public class CustomSerializer implements Serializer<MessageDTO> {
```

Comment:
why do not use standart `org.springframework.kafka.support.serializer.JsonSerializer`? do not see specific configuration

Reply:
OK. This is better solution. Change made